cli/src/commands/sign.rs

```diff
@@ -0,0 +1,71 @@
+use std::io::Write;
+
+use jj_lib::backend::ObjectId;
+use jj_lib::signing::SignBehavior;
+
+use crate::cli_util::{user_error, CommandError, CommandHelper, RevisionArg};
+use crate::ui::Ui;
+
+/// Cryptographically sign a revision
+#[derive(clap::Args, Clone, Debug)]
+pub struct SignArgs {
+    /// What key to use, depends on the configured signing backend.
+    #[arg()]
+    key: Option<String>,
+    /// What revision to sign
+    #[arg(long, short, default_value = "@")]
+    revision: RevisionArg,
+    /// Sign the commit that is not authored by you or was already signed.
+    #[arg(long, short)]
+    force: bool,
+    /// Drop the signature, explicitly "un-signing" the commit.
+    #[arg(long, short = 'D', conflicts_with = "force")]
+    drop: bool,
```

Comment:
nit: might be better to make `sign --drop` a separate command. `jj` tends to have create/delete/set/get/list verbs as subcommands.